pyo3-bytes/src/bytes.rs

```diff
@@ -396,3 +594,36 @@ impl<'py> FromPyObject<'py> for PyBytesWrapper {
         Ok(Self(Some(buffer)))
     }
 }
+
+/// Returns a string like `b"asdf\n\0"` for the given byte slice.
+fn python_bytes_repr(data: &[u8]) -> String {
+    let mut out = String::from("b\"");
+    for &byte in data {
+        match byte {
+            b'\\' => out.push_str("\\\\"),
```

Comment:
Also, instead of adding these extra `\`, we could just put `br` in the repr. I.e. these are equal:
```py
br"\n" == b"\\n"
```
So the repr can be
```
Bytes(br"\n")
```

Reply:
wont that not matter if we use the bytes debug impl?

Reply:
We need the `r` raw string to ensure the bytes debug repr is correct